src/depends/libDatabase/LevelDB.cpp

```diff
@@ -87,21 +82,13 @@ LevelDB::LevelDB(const string & dbName, const string & subdirectory)
     leveldb::DB* db;
     leveldb::Status status;
 
-    if(m_subdirectory.empty())
+    string db_path = diagnostic ? ((m_subdirectory + (m_subdirectory.back() == '/' ? "" : "/")) + PERSISTENCE_PATH) : ("./"  + PERSISTENCE_PATH + (m_subdirectory.empty() ? "" : "/" + m_subdirectory));
```

Comment:
To simplify this long statement, maybe remove the need for checking `m_subdirectory.back` by making sure getnetworkhistory removes any trailing slash before calling this constructor.

Also, might be good to put a comment here about why we handle diagnostic differently.  You can say that the bool is only ever true when used by the tool, and that the non-diagnostic path is preserved from the original code.

Reply:
Done!